src/compiler/set.h

```diff
@@ -101,13 +101,30 @@ template<typename T> class UnorderedSet {
   }
   bool erase(T x) { return set_.erase(x) > 0; }
 
+  template<typename F>
+  void erase_if(const F& callback) {
```

Comment:
I think `erase_if` sounds a bit too much like erase-everything-if-the-predicate-returns-true-when-called-once. Maybe `erase_where` or `erase_all_where`?

```suggestion
  void erase_all_where(const F& callback) {
```